Library/Homebrew/os/linux/global.rb

```diff
@@ -1,7 +1,7 @@
 # frozen_string_literal: true
 
 # enables experimental readelf.rb, patchelf support.
-HOMEBREW_PATCHELF_RB = ENV["HOMEBREW_PATCHELF_RB"].present?.freeze
+HOMEBREW_PATCHELF_RB = (ENV["HOMEBREW_DEVELOPER"].present? && ENV["HOMEBREW_PATCHELF_RB"] != "0").freeze
```

Comment:
I think this assignment currently misses the case where `HOMEBREW_DEVELOPER` is not set, but `HOMEBREW_PATCHELF_RB` is set. We should handle that case.

In other words, here are our exhaustive conditions:

* `HOMEBREW_DEVELOPER` and `HOMEBREW_PATCHELF_RB != 0`: enable patchelf.rb
* `HOMEBREW_DEVELOPER` and `HOMEBREW_PATCHELF_RB == 0`: disable patchelf.rb
* No `HOMEBREW_DEVELOPER` and `HOMEBREW_PATCHELF_RB != 0`: enable patchelf.rb
* No `HOMEBREW_DEVELOPER` and `HOMEBREW_PATCHELF_RB == 0` *or* not present: disable patchelf.rb